src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseTodoListTable.cs

```diff
@@ -303,19 +303,38 @@ private LinePosition GetLineColumn(TodoItem item)
 
                 public override bool TryNavigateTo(int index, bool previewTab)
                 {
-                    var item = GetItem(index)?.Primary;
+                    var item = GetItem(index);
                     if (item == null)
                     {
                         return false;
                     }
 
-                    var trackingLinePosition = GetTrackingLineColumn(item.Workspace, item.DocumentId, index);
-                    if (trackingLinePosition != LinePosition.Zero)
+                    var primary = item.Primary;
+
+                    var workspace = item.Workspace;
+                    var solution = workspace.CurrentSolution;
+                    var document = solution.GetDocument(primary.DocumentId);
+                    if (document == null)
+                    {
+                        return false;
+                    }
+
+                    int line, column;
+                    LinePosition trackingLinePosition;
+
+                    if (workspace.IsDocumentOpen(document.Id) &&
+                        (trackingLinePosition = GetTrackingLineColumn(document, index)) != LinePosition.Zero)
+                    {
+                        line = trackingLinePosition.Line;
+                        column = trackingLinePosition.Character;
+                    }
+                    else
                     {
-                        return TryNavigateTo(item.Workspace, item.DocumentId, trackingLinePosition.Line, trackingLinePosition.Character, previewTab);
+                        line = primary.OriginalLine;
+                        column = primary.OriginalColumn;
                     }
```

Comment:
this logic loos very duplicative with what's two files above.  No way to share more?

Reply:
I think we can, but needs a bit more refactoring. I'll follow up.

---
In reply to: [287435304](https://github.com/dotnet/roslyn/pull/35902#discussion_r287435304) [](ancestors = 287435304)

Reply:
Done.